common/components/register_model/spec.yaml

```diff
@@ -52,12 +59,17 @@ inputs:
     optional: false
     description: Path to the model directory
 
-  model_info_path:
+  download_details:
+    type: uri_file
+    optional: true
+    description: A JSON file which contains information related to model download.
+
+  model_metadata:
```

Comment:
@ManojBableshwar can you please share your recommendation here? `model_metadata` to be used for capturing information model information in YAML, some of which can other way be provided using component simple inputs as well. model_name, model_description to consume model card in the description and information such as model flavor, tags and properties which can be passed using a YAML file